src/urllib3/connectionpool.py

```diff
@@ -386,10 +387,23 @@ def _make_request(
 
         # conn.request() calls httplib.*.request, not the method in
         # urllib3.request. It also calls makefile (recv) on the socket.
-        if chunked:
-            conn.request_chunked(method, url, **httplib_request_kw)
-        else:
-            conn.request(method, url, **httplib_request_kw)
+        try:
+            if chunked:
+                conn.request_chunked(method, url, **httplib_request_kw)
+            else:
+                conn.request(method, url, **httplib_request_kw)
+
+        # We are swallowing BrokenPipeError (errno.EPIPE) since the server is
+        # legitimately able to close the connection after sending a valid response.
+        # With this behaviour, the received response is still readable.
+        except BrokenPipeError:  # Python 3
+            pass
+        except IOError as e:
+            # https://erickt.github.io/blog/2014/11/19/adventures-in-debugging-a-potential-osx-kernel-bug/
+            if e.errno == errno.EPROTOTYPE:  # macOS
+                pass
+            elif e.errno != errno.EPIPE:  # Python 2
```

Comment:
We can simplify this a bit to `if e.errno not in (errno.EPROTOTYPE, errno.EPIPE): raise`

Reply:
Yeah, I did it like this to prepare for Python 2 removal, but happy to merge those as long as we still say "Python 2" and "macOS" around those lines

Reply:
That's fair too, it took me a moment to verify that we were doing the same thing for the `BrokenPipeError` and the `e.errno==EPIPE/EPROTOTYPE` case so wanted to prevent that from happening to others.

Reply:
Oh, I understand now. BrokenPipeError is actually not only documented as EPIPE but also ESHUTDOWN, and I failed at replacing it with errno checks, but didn't try too hard.